### PR TITLE
Add coverage workflow

### PR DESCRIPTION
**Pull request by Thoralf-M**

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        # Restore from outdated cache for speed
+        restore-keys: |
+          ubuntu-latest-nightly-cargo-index-${{ hashFiles('**/Cargo.lock') }}
+          ubuntu-latest-nightly-cargo-index-
+
+    - name: Install binutils and rustfilt
+      run: cargo install cargo-binutils rustfilt
+
+    - name: Run tests and collect coverage
+      working-directory: ./client
```
Comment (thibault-martinez) [inline .github/workflows/coverage.yml]: Can't we do coverage of the whole repository ? We surely want types to be covered as well

Comment (Thoralf-M) [reply .github/workflows/coverage.yml]: That's done, see https://coveralls.io/github/Thoralf-M/iota.rs
But when I moved the coverage.sh file to the root it returned 0 as coverage, so I moved it back there :shrug:

Comment (thibault-martinez) [reply .github/workflows/coverage.yml]: Oof that's going to trigger me

Comment (Thoralf-M) [reply .github/workflows/coverage.yml]: This was the run where it returned 0, do you see anything wrong there? https://github.com/Thoralf-M/iota.rs/actions/runs/3712563503/jobs/6294235150

Comment (thibault-martinez) [reply .github/workflows/coverage.yml]: Hmm no not really

Comment (thibault-martinez) [reply .github/workflows/coverage.yml]: I'll have a look at coverage.sh later, there has to be something with the fact that ./ is a workspace

File: client/coverage.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+rm -rf coverage
+mkdir coverage
+
+# Run tests with profiling instrumentation
+echo "Running instrumented unit tests..."
+RUSTFLAGS="-C instrument-coverage" LLVM_PROFILE_FILE="iota-client-%m.profraw" cargo +nightly test --all --all-features
+
+# Merge all .profraw files into "iota-client.profdata"
+echo "Merging coverage data..."
+cargo +nightly profdata -- merge ./iota-client-*.profraw -o iota-client.profdata
```
Comment (kwek20) [inline client/coverage.sh]: probably here it tries to find this data but it cant when we have it in the workspace?

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (Thoralf-M) [reply .github/workflows/coverage.yml]: All is covered now https://coveralls.io/github/Thoralf-M/iota.rs

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**